Sources/Fullscreen/FavoriteAdsList/FavoriteAdsListView.swift

```diff
@@ -0,0 +1,246 @@
+    func favoriteAdsListView(_ view: FavoriteAdsListView, titleForHeaderInSection section: Int) -> String?
+    func favoriteAdsListView(_ view: FavoriteAdsListView, viewModelFor indexPath: IndexPath) -> FavoriteAdViewModel
+    func favoriteAdsListView(_ view: FavoriteAdsListView,
+                             loadImageWithPath imagePath: String,
+                             imageWidth: CGFloat,
+                             completion: @escaping ((UIImage?) -> Void)
+    )
+    func favoriteAdsListView(_ view: FavoriteAdsListView,
+                             cancelLoadingImageWithPath imagePath: String,
+                             imageWidth: CGFloat
+    )
+}
+
+public class FavoriteAdsListView: UIView {
+
+    // MARK: - Public properties
+
+    public weak var delegate: FavoriteAdsListViewDelegate?
+    public weak var dataSource: FavoriteAdsListViewDataSource?
+
+    public var title: String = "" {
+        didSet { tableHeaderView.title = title }
+    }
+
+    public var subtitle: String = "" {
+        didSet { tableHeaderView.subtitle = subtitle }
+    }
+
+    public var searchBarPlaceholder: String = "" {
+        didSet { tableHeaderView.searchBarPlaceholder = searchBarPlaceholder }
+    }
+
+    public var searchBarText: String {
+        get { return tableHeaderView.searchBarText }
+        set { tableHeaderView.searchBarText = searchBarText }
+    }
+
+    public var sortingTitle: String = "" {
+        didSet { tableHeaderView.sortingTitle = sortingTitle }
+    }
+
+    // MARK: - Private properties
+
+    private let imageCache = ImageMemoryCache()
+    private var didSetTableHeader = false
+
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(withAutoLayout: true)
+        tableView.register(FavoriteAdTableViewCell.self)
+        tableView.register(FavoriteAdsSectionHeaderView.self)
+        tableView.tableFooterView = UIView()
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorInset = .leadingInset(frame.width)
+        tableView.keyboardDismissMode = .onDrag
+        return tableView
+    }()
+
+    private lazy var tableHeaderView: FavoriteAdsListTableHeader = {
+        let tableHeader = FavoriteAdsListTableHeader(withAutoLayout: true)
+        tableHeader.delegate = self
+        tableHeader.searchBarDelegate = self
+        return tableHeader
+    }()
+
+    // MARK: - Init
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) { fatalError() }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(tableView)
+        tableView.fillInSuperview()
+    }
+
+    // MARK: - Overrides
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if !didSetTableHeader {
+            tableView.tableHeaderView = tableHeaderView
+
+            NSLayoutConstraint.activate([
```

Comment:
Why do we need to do it in `layoutSubviews` and not in `setup`?

Reply:
Because I experienced that the header jumped around if I placed it there. Doing autolayout for `tableView.tableHeaderView` is hard